webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPDirectiveRepository.scala

```diff
@@ -703,6 +702,46 @@ class WoLDAPDirectiveRepository(
     })
   }
 
+  /**
+   * Delete a directive's system.
+   * No dependency check are done, and so you will have to
+   * delete dependent rule (or other items) by
+   * hand if you want.
+   *
+   * If no directive has such id, return a success.
+   */
+  def deleteSystemDirective(id:DirectiveId, modId: ModificationId, actor:EventActor, reason:Option[String]) : IOResult[Option[DeleteDirectiveDiff]] = {
+    getActiveTechniqueAndDirectiveEntries(id).flatMap {
```

Comment:
can you create a private methods with the common code, and a "onlySystem" boolean parameter that only call the common method with the corresponding error check ? We should never duplicate that much code, especially it it's so much similar.